src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -180,62 +178,102 @@ private abstract class ReadOperation : AsyncOperation
 
         private abstract class WriteOperation : AsyncOperation 
         {
-        }        
+        }
 
-        private sealed class SendOperation : WriteOperation
+        private abstract class SendOperation : WriteOperation
         {
-            public byte[] Buffer;
-            public int Offset;
-            public int Count;
             public SocketFlags Flags;
             public int BytesTransferred;
-            public IList<ArraySegment<byte>> Buffers;
-            public int BufferIndex;
+            public int Offset;
+            public int Count;
 
             protected sealed override void Abort() { }
 
             public Action<int, byte[], int, SocketFlags, SocketError> Callback
             {
-                private get { return (Action<int, byte[], int, SocketFlags, SocketError>)CallbackOrEvent; }
-                set { CallbackOrEvent = value; }
+                set => CallbackOrEvent = value;
```

Comment:
This seems like a good change, but we should be consistent about doing it this way across all operations.  I'm hacking on this code and can do this in a subsequent change.